### PR TITLE
Add floating point support

### PR DESCRIPTION
**Pull request by casuallyblue**
This is the initial steps of floating point support for mirth. I'm uploading it before it is completely done for feedback before I implement any further. This includes a partial implementation of a mandelbrot set renderer example.

### CODE REVIEW

**Comment by typeswitch-dev** [comment]
Great work! The only note I have is that I'd like the name of the float type in mirth be:

- `Float64` for double-precision floats, and
- `Float32` for single-precision floats.

Once that is renamed, I think it would be ok to merge this PR, even without single-precision!

**Comment by casuallyblue** [comment]
> Great work! The only note I have is that I'd like the name of the float type in mirth be:
> 
>     * `Float64` for double-precision floats, and
> 
>     * `Float32` for single-precision floats.
> 
> 
> Once that is renamed, I think it would be ok to merge this PR, even without single-precision!

Everything's renamed, and I made the internal value be a float instead of storing as a string. Still figuring out why the windows build suddenly can't find `snprintf` even though it could before.

File: src/mirth.h [hunk 1]
```diff
@@ -39,6 +40,7 @@ extern int write(int, const char*, size_t);
 extern int close(int);
 extern int open(const char*, int, ...);
 extern void exit(int);
+extern int snprintf (char * s, size_t n, const char * format, ...);
```
Comment (typeswitch-dev) [inline src/mirth.h:43]:
```suggestion
extern int sprintf (char * s, const char * format, ...);
```

File: src/mirth.h [hunk 2]
```diff
@@ -720,6 +761,67 @@ static void mp_primZ_strZ_cmp (void) {
...
+	push_f64(a / b);
+	PRIM_EXIT(mp_primZ_float64Z_div);
+}
+
+static void mp_primZ_float64Z_toZ_str (void) {
+	PRIM_ENTER(mp_primZ_float64Z_toZ_str, "prim-float64-to-str");
+	double d = pop_f64();
+	int len = snprintf(NULL, 0, "%.*f", DBL_DIG, d);
+	char* result = malloc(len+1);
+	snprintf(result, len+1, "%.*f", DBL_DIG,  d);
```
Comment (typeswitch-dev) [inline src/mirth.h:819]:
```suggestion
	char result[DBL_DIG+32] = {0};
	int len = sprintf(result,"%.*g", DBL_DIG,  d);
```

**Review by typeswitch-dev — COMMENTED** [review]
maybe we can just use sprintf here (temporarily)

**Comment by typeswitch-dev** [comment]
Thanks @casuallyblue!

**Pull request merged**